### DISCUSSION TITLE
what is the meaning of null childblock?

### DISCUSSION THREAD

**Discussion by OriLiMu** [ScratchAddons/ScratchAddons#7596] [Help] [answered]
The block is like this. I want to know the block structure of it. 
![](https://raw.githubusercontent.com/woshichuanqilz/OriImg/master/20240626114617.png)

As you can see in the image, the first `repeat_10` block should have 2 children, `repeat_10`(second) and the `say_hello` block. 

But in the debug Info I see 3 children blocks. The first child block's category is `null`. Why is that? What is the meaning of this null category block

![image](https://github.com/ScratchAddons/ScratchAddons/assets/11295355/514edf68-e215-4cbc-a8b6-f5fc2bfb294d)

**Comment by mxmou** [comment, marked as answer]
It's the number input. Note that its color is white and its `type` is `math_whole_number`.
